### PR TITLE
CHA-RL3j: release on initialized status - re-order for continuity

### PR DESCRIPTION
**Pull request by AndyTWF**
Re-orders the spec points around CHA-RL3j to provide better continuity when reading.

[CHA-716]

[CHA-716]: https://ably.atlassian.net/browse/CHA-716?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: textile/chat-features.textile [hunk 1]
```diff
@@ -115,6 +115,7 @@ h4(#rooms-lifecycle-operations). Room Lifecycle Operations
 * @(CHA-RL3)@ A room must be explicitly released via the @RELEASE@ operation. This operation is not performed directly on a Room object by the client, but is described here.
 ** @(CHA-RL3a)@ @[Testable]@ If the room is already in the @RELEASED@ status, this operation is no-op.
 ** @(CHA-RL3b)@ @[Testable]@ If the room is in the @DETACHED@ status, then the room is immediately transitioned to @RELEASED@ and the operation succeeds.
+** @(CHA-RL3m)@ @[Testable]@ If the room is in the @INITIALIZED@ status, then the room is immediately transitioned to @RELEASED@ and the operation succeeds.
```
Comment (sacOO7) [inline textile/chat-features.textile:118]: We already have spec for this CHA-RL3j, we don't need to update this 🤪

Comment (AndyTWF) [reply textile/chat-features.textile:118]: I've moved the spec points around a bit so the removed one (3i) is at the bottom of the block, to provide better continuity as I'd expect 3j to sit immediately below 3b.

**Review by sacOO7 — APPROVED** [review]
LGTM

**Comment by sacOO7** [comment]
@AndyTWF Btw, might not be related to this PR, but I needed more clarification for 
[CHA-RL3h] - Upon operation completion, the underlying Realtime Channels are released from the core SDK prevent leakage.
I couldn't find any references for the same in code/tests for chat-js SDK. 
The only thing we do as a part of `room-release` is make sure channels go into either detached or failed state.

**Comment by AndyTWF** [comment]
> @AndyTWF Btw, might not be related to this PR, but I needed more clarification for [CHA-RL3h] - Upon operation completion, the underlying Realtime Channels are released from the core SDK prevent leakage. I couldn't find any references for the same in code/tests for chat-js SDK. The only thing we do as a part of `room-release` is make sure channels go into either detached or failed state.

It's a bit oddly placed, but it's here: https://github.com/ably/ably-chat-js/blob/0e1b52336a9e15c711d62848a0c3ba841dc6ae77/src/core/room.ts#L203-L218

Test here: https://github.com/ably/ably-chat-js/blob/0e1b52336a9e15c711d62848a0c3ba841dc6ae77/test/core/room.test.ts#L183-L214

**Comment by sacOO7** [comment]
Okay, this kinda comes under a new world we were talking about. Ideally, should have been part of room-lifecycle-manager-> mutex. But we placed it here since `lifecycle-manager` doesn't have access to `realtime.channels` : (

**Comment by AndyTWF** [comment]
> Okay, this kinda comes under a new world we were talking about. Ideally, should have been part of room-lifecycle-manager-> mutex. But we placed it here since `lifecycle-manager` doesn't have access to `realtime.channels` : (

From a specification point of view - it can absolutely go in the lifecycle manager (I'm not sure why we didn't in JS, if I'm honest).

**Comment by sacOO7** [comment]
Okay, found the [`channels.release` method](https://github.com/ably/ably-js/blob/18a255948c38d1e60715c8f5d6173369b57cb8d6/src/common/lib/client/baserealtime.ts#L218). Seems, you can call `channels.release` on `channel` object itself
```
channel.client.channels.release(channel.name)
``` 
So, https://github.com/ably/ably-chat-js/blob/0e1b52336a9e15c711d62848a0c3ba841dc6ae77/src/core/room-lifecycle-manager.ts#L891 can be updated accordingly.

**Comment by AndyTWF** [comment]
> Okay, found the [`channels.release` method](https://github.com/ably/ably-js/blob/18a255948c38d1e60715c8f5d6173369b57cb8d6/src/common/lib/client/baserealtime.ts#L218). Seems, you can call `channels.release` on `channel` object itself
> 
> ```
> channel.client.channels.release(channel.name)
> ```
> 
> So, https://github.com/ably/ably-chat-js/blob/0e1b52336a9e15c711d62848a0c3ba841dc6ae77/src/core/room-lifecycle-manager.ts#L891 can be updated accordingly.

There isn't a method that lets you call release on the actual channel object - you have to call it on `realtime.channels`, so still need the `realtime` client available, or at least its channel map.

**Comment by sacOO7** [comment]
That's what I meant, [you can access `realtime` client from `channel` itself](https://github.com/ably/ably-js/blob/18a255948c38d1e60715c8f5d6173369b57cb8d6/src/common/lib/client/realtimechannel.ts#L108)
```
BaseRealtime client = channel.client
```
So 
```
channel.client.channels.release(channel.name)
```
is possible.

**Comment by AndyTWF** [comment]
> That's what I meant, [you can access `realtime` client from `channel` itself](https://github.com/ably/ably-js/blob/18a255948c38d1e60715c8f5d6173369b57cb8d6/src/common/lib/client/realtimechannel.ts#L108)
> 
> ```
> BaseRealtime client = channel.client
> ```
> 
> So
> 
> ```
> channel.client.channels.release(channel.name)
> ```
> 
> is possible.

I don't think that's exported by `ably-js`? At least, I can't see it in the `RealtimeChannel` def in `ably.d.ts`.

**Comment by sacOO7** [comment]
Yeah, seems not exported, but `client` object is surely accessible
```
(channel : any | { client: RealtimeClient }).client.channels.release(channel.name)
```

**Comment by sacOO7** [comment]
Btw, thanks for the info. Will implement accordingly in kotlin 👍

**Review by splindsay-92 — APPROVED** [review]

**Pull request merged**